lib/grandpa/finalization.go

```diff
@@ -0,0 +1,493 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+)
+
+type ephemeralService interface {
+	Start() error
+	Stop() error
+}
+
+type finalizationHandler struct {
+	servicesLock       sync.Mutex
```

Comment:
I disagree, otherwise it exposes the Lock and Unlock methods to the caller.

Also very niche complain, but it's annoying you cannot CTRL+Click on the embedded mutex when finding a `.Lock()` call.

Reply:
I think the named field makes explicit that the locking purpose is to lock the ephemeral services we create each round